File: doc/codes/python/examples/api/compute/pattern/plane.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3_pattern import plane, wavelength as calc_wavelength

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

direction = np.array([0.0, 0.0, 1.0])
wavelength = calc_wavelength(340 * m / s)
dst = geometry.pattern_buffer()

# ANCHOR: api
plane(geometry, direction, wavelength, dst)
# ANCHOR_END: api

File: doc/codes/rust/examples/api/compute/pattern/plane.rs
use autd3_rs::geometry::{Autd3, Geometry, Vector3};
use autd3_rs::units::{m, s};
use autd3_rs_pattern::{plane, wavelength};

fn main() {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let direction = Vector3::z_axis();
    let wavelength = wavelength(340.0 * m / s);
    let mut dst = geometry.pattern_buffer();

    // ANCHOR: api
    plane(&geometry, direction, wavelength, &mut dst);
    // ANCHOR_END: api
}
